src/bio.c

```diff
@@ -361,33 +360,33 @@ void *bioProcessBackgroundJobs(void *arg) {
 
         /* Lock again before reiterating the loop, if there are no longer
          * jobs to process we'll block again in pthread_cond_wait(). */
-        pthread_mutex_lock(&bio_mutex[worker]);
-        listDelNode(bio_jobs[worker], ln);
+        pthread_mutex_lock(&bwd->bio_mutex);
+        listDelNode(bwd->bio_jobs, ln);
         bio_jobs_counter[job_type]--;
-        pthread_cond_signal(&bio_newjob_cond[worker]);
+        pthread_cond_signal(&bwd->bio_newjob_cond);
     }
 }
 
 /* Return the number of pending jobs of the specified type. */
 unsigned long bioPendingJobsOfType(int type) {
-    unsigned int worker = bio_job_to_worker[type];
+    bio_worker_data *const bwd = &bio_workers[bio_job_to_worker[type]];
```

Comment:
Shouldn't `const` be for data, not for pointer?
```suggestion
    const bio_worker_data *bwd = &bio_workers[bio_job_to_worker[type]];
```

Reply:
No, it's the pointer that will not change. The actual data must be mutable since
`pthread_mutex_lock` / `pthread_mutex_unlock` requires a `pthread_mutex_t *` pointer, not a `const pthread_mutex_t *` and they will actually make changes to the mutex state - hence the `bio_worker_data` state. Your suggested change would imply that `bwd` can be changed to point somewhere else in the loop, which it will not, but it also means that noone is allowed to mutate the data to which `bwd` points, so compilation would fail.